src/main/inAppNotifications/useInAppNotifications.ts

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useMemo } from 'react';
+import {
+  CalloutType,
+  CommunityContributorType,
+  InAppNotificationCategory,
+  InAppNotificationState,
+  NotificationEventType,
+  PlatformRole,
+  SpaceLevel,
+} from '@/core/apollo/generated/graphql-schema';
+import {
+  refetchInAppNotificationsQuery,
+  useInAppNotificationsQuery,
+  useUpdateNotificationStateMutation,
+} from '@/core/apollo/generated/apollo-hooks';
+import { useUserContext } from '@/domain/community/user';
+
+const POLLING_INTERVAL = 5 * 1000; // 5 seconds
+
+export interface InAppNotificationProps {
+  id: string;
+  type: NotificationEventType;
+  triggeredAt: Date;
+  state: InAppNotificationState;
+  category: InAppNotificationCategory;
+  triggeredBy?: {
+    profile:
+      | {
+          displayName: string;
+          url: string;
+          visual?: {
+            uri: string;
+          };
+        }
+      | undefined;
+  };
+  contributor?: {
+    type: CommunityContributorType;
+    profile:
+      | {
+          displayName: string;
+          url: string;
+          visual?: {
+            uri: string;
+          };
+        }
+      | undefined;
+  };
+  actor?: {
+    __typename: string;
+    profile:
+      | {
+          displayName: string;
+          url: string;
+          visual?: {
+            uri: string;
+          };
+        }
+      | undefined;
+  };
+  callout?: {
+    type: CalloutType;
+    framing:
+      | {
+          profile?:
+            | {
+                displayName: string;
+                url: string;
+              }
+            | undefined;
+        }
+      | undefined;
+  };
+  space?: {
+    level: SpaceLevel;
+    profile:
+      | {
+          displayName: string;
+          url: string;
+          visual?: {
+            uri: string;
+          };
+        }
+      | undefined;
+  };
+  comment?: string;
+  commentUrl?: string;
+}
+
+export const useInAppNotifications = () => {
+  const { user, platformRoles } = useUserContext();
+
+  const [updateState] = useUpdateNotificationStateMutation();
+
+  const enableNotifications = useMemo(() => {
```

Comment:
I think this should go into the `InAppNotificationsContext`. 
It may be a bigger refactor than I think, but this context sits under the UserContext so it should have no problem to access these, and this way we'll unify the place where we change if notifications are enabled for this user or not... Does that make sense? disregard if it's too big refactor, I've only seen this check twice for now

Reply:
It doesn't require big refactoring. Let me think about it conceptually as on first thought, I'd like to keep the context as independent as possible.

Reply:
I rethought it and I agree that this suggestion is the right thing to do, applying separation of concerns, consistency, and DRY. Performance-wise it's not ideal, but it's not something critical either. Thanks, changes applied, please double-check.